solidity/src/contracts/Adapter.sol

```diff
@@ -24,7 +24,8 @@ contract Adapter is IAdapter, Ownable, ReentrancyGuard {
         ); // swap event custom topic0
 
     uint256 public nonce;
-    address public immutable erc20;
+    bytes32 public immutable erc20;
+    address public immutable erc20Bytes;
```

Comment:
I would keep just the bytes, the address will be computed in place where needed
```suggestion
    bytes32 public immutable erc20;
```

Reply:
Also I'm not sure why you named the variable `erc20Bytes` and store it as an address, not much intuitive I'd say

Reply:
Ah I've understood why you did it, I think it's good, but I'd change the name according to the type
```js
address erc20;
bytes32 erc20Bytes;
```

Reply:
In an analogous way to what done in Antelope contracts I keep the erc20 token both in a address (name for antelope) form and bytes form, so that an underlying coming from a non EVM (Antelope) chain is supported.

I used the wrong name though, as it should be address erc20Addr.
As discussed privately I therefore propose the following name scheme:
```
bytes32 erc20;
address erc20Addr;
```